components/downgrade_feedback_modal/index.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import {useDispatch} from 'react-redux';
+import classNames from 'classnames';
+
+import {GenericModal} from '@mattermost/components';
+import {DowngradeFeedback} from '@mattermost/types/cloud';
+import {closeModal} from 'actions/views/modals';
+import RadioButtonGroup from 'components/common/radio_group';
+
+import {ModalIdentifiers} from 'utils/constants';
+
+import './index.scss';
+
+type Props = {
+    onSubmit: (downgradeFeedback: DowngradeFeedback) => void;
+}
+
+export default function DowngradeFeedbackModal(props: Props) {
+    const optionOther = 'Other';
+    const feedbackOptions: string[] = [
+        'Experienced technical issues',
+        'No longer need Cloud Professional features',
+        'Exploring other solutions',
+        'Too expensive',
+        optionOther,
+    ];
+
+    const [reason, setReason] = useState('');
+    const [comments, setComments] = useState('');
+    const reasonNotSelected = reason === '';
+    const commentsNotProvided = comments === '';
+    const downgradeDisabled = reasonNotSelected || (reason === optionOther && commentsNotProvided);
+
+    const dispatch = useDispatch();
+
+    const handleSubmitFeedback = () => {
+        if (reason === optionOther && !comments) {
+            return;
+        }
+
+        props.onSubmit({reason, comments});
+        dispatch(closeModal(ModalIdentifiers.DOWNGRADE_FEEDBACK));
+    };
+
+    const handleCancel = () => {
+        dispatch(closeModal(ModalIdentifiers.DOWNGRADE_FEEDBACK));
+    };
+
+    return (
+        <GenericModal
+            onExited={handleCancel}
+            className='DowngradeFeedback__Container'
+        >
+            <span className='DowngradeFeedback__Title'>
+                <FormattedMessage
+                    id={'downgrade_feedback.title'}
+                    defaultMessage={'Please share your reason for downgrading'}
+                />
+            </span>
+            <RadioButtonGroup
+                id='downgradeFeedbackRadioGroup'
+                testId='downgradeFeedbackRadioGroup'
+                values={feedbackOptions.map((option) => {
+                    return {
+                        value: option,
+                        key: option,
+                        testId: option,
+                    };
+                })}
+                value={reason}
+                onChange={(e) => setReason(e.target.value)}
+            />
+            <textarea
+                className='DowngradeFeedback__FreeFormText'
+                placeholder='Please tell us why you are downgrading...'
+                rows={3}
+                onChange={(e) => {
+                    setComments(e.target.value.trim());
+                }}
+                style={{display: reason === optionOther ? '' : 'none', resize: 'none'}}
```

Comment:
Since this has a change handler but doesn't take its value from the state value, this is effectively an [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html). It works, but is less idiomatic. To get the effect of only submitting the trimmed data to the server, you could `.trim()` the value when checking `commentsNotProvided` and when you call props.onSubmit, e.g. `props.onSubmit({reason, comments: comments.trim()});`